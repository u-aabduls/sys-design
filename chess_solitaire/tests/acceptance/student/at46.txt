-- ACCEPTANCE TEST #46: VIOLATION CASE: QUEEN BLOCKED - #3

-- Setup some chess pieces for the game
setup_chess (Q,1,3)
setup_chess (K,2,3)
setup_chess (B,3,3)
setup_chess (Q,4,3)

start_game

-- Blocked by King@(2,3)
move_and_capture(1,3,4,3)

-- Blocked by King@(2,3)
move_and_capture(1,3,3,3)

