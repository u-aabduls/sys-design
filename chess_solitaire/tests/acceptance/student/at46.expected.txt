  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(Q,1,3)
  # of chess pieces on board: 1
  Game being Setup...
  ..Q.
  ....
  ....
  ....
->setup_chess(K,2,3)
  # of chess pieces on board: 2
  Game being Setup...
  ..Q.
  ..K.
  ....
  ....
->setup_chess(B,3,3)
  # of chess pieces on board: 3
  Game being Setup...
  ..Q.
  ..K.
  ..B.
  ....
->setup_chess(Q,4,3)
  # of chess pieces on board: 4
  Game being Setup...
  ..Q.
  ..K.
  ..B.
  ..Q.
->start_game
  # of chess pieces on board: 4
  Game In Progress...
  ..Q.
  ..K.
  ..B.
  ..Q.
->move_and_capture(1,3,4,3)
  # of chess pieces on board: 4
  Error: Block exists between (1, 3) and (4, 3)
  ..Q.
  ..K.
  ..B.
  ..Q.
->move_and_capture(1,3,3,3)
  # of chess pieces on board: 4
  Error: Block exists between (1, 3) and (3, 3)
  ..Q.
  ..K.
  ..B.
  ..Q.
